pyiron_base/jobs/job/extension/files.py

```diff
@@ -95,15 +95,18 @@ def __getitem__(self, item):
         if item not in _working_directory_list_files(
             working_directory=self._working_directory
         ):
-            raise KeyError(item)
+            raise FileNotFoundError(item)
 
         return File(os.path.join(self._working_directory, item))
 
     def __getattr__(self, item):
-        try:
-            return self[self._get_file_dict()[item]]
-        except KeyError:
-            raise AttributeError(item) from None
+        if item == "__dict__":
```

Comment:
I suppose this check is because sometimes python internals try to check whether we define dunder vars, but they would get confused if met with a `FileNotFoundError`? Is `__dict__` the only that can happen?  Maybe
```suggestion
        if item.startswith("__") and item.endswith("__"):
```
is safer.

Reply:
> Is `__dict__` the only that can happen? 

The only thing I found during the tests, it seems the `dir()` method checks for `__dict__` and expects to get an `AttributeError` if it is not defined.